### DISCUSSION TITLE
nrf51 chip not booting when bluetooth stack added

### DISCUSSION THREAD

**Discussion by ThinkTransit** [micropython/micropython#15204] [nRF]
I have some bluetooth modules I got from aliexpress that have a NRF51822 chip inside.

Because they have different pins exposed to existing boards I have a custom board definition that defines the repl.

```
// UART config
#define MICROPY_HW_UART1_RX         (12)
#define MICROPY_HW_UART1_TX         (13)
```

When I build the firmware with the following command, everything works as expected, I can access the repl through pins 12/13

```
make BOARD_DIR=/home/patrick/esp/micropython-ia/ports/nrf/boards/IA-BEACON 
```

However when I download the BT stack and compile with 

```
make BOARD_DIR=/home/patrick/esp/micropython-ia/ports/nrf/boards/IA-BEACON SD=s110
```

It get nothing on the serial pins.

Not sure if adding the SD param is interfering with my custom repl pins or it's just not booting for some other reason.

Any ideas? I'm using an STLINK and openocd to "halt" "program" "reset" if that makes a difference!

**Comment by andrewleech** [comment]
Double check the 32KHz crystal build settings, I ran into that in the past a fair bit on why the SD can cause lockups at boot.

**Reply by ThinkTransit** [reply]
Thanks Andrew any pointers where I can find the crystal build settings?

**Reply by andrewleech** [reply]
Try browsing the other boards files for a line like this I think (it's been years since I used a softdevice though so I'm not certain): https://github.com/micropython/micropython/blob/master/ports%2Fnrf%2Fboards%2FWT51822_S4AT%2Fmpconfigboard.mk#L7

The correct setting will depend on what clock sources your hardware has.

**Reply by ThinkTransit** [reply]
Thanks again I will dig into that!
